### ISSUE TITLE
[a11y]: In ErrorBoundary component once the message initiated the same message is not announced instantaneously

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16287]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

1.55.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

In the `ErrorBoundary` component, whether with default settings or custom context, 
the warning message below the button should not be repeated instantly once it's initiated.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-errorboundary--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
